### PR TITLE
cmd: make explain_test can test collation-related tests with new-collation disabled

### LINKED ISSUE #31384: Enhance explain_test to make it test collation-related tests with new-collation disabled.
## Enhancement
We want to make explain_test can test collation-related tests with new-collation disabled.

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #31384

Problem Summary:

### What is changed and how it works?
1. The test involves collation, and the test files are prefixed with collation.
2. The default configuration is new collation, and another configuration turns off the new collation.
3. After the default test runs all tests, use the configuration of disabling new collation to run collation-related tests.
4. The results of tests with new collation open have the suffix enabled and closed have the suffix disabled.
For example collation_test1, the result file is collation_test1_enabled, collation_test1_disabled
5. In the explaintest main.go. Modify resultFileName() to control the collation result file
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: cmd/explaintest/disable_new_collation.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+host = "127.0.0.1"
+
+[status]
+status-host = "127.0.0.1"
+
+[performance]
+stats-lease = "0"
+
+[experimental]
+enable-new-charset = true
```
Comment (tangenta) [inline cmd/explaintest/disable_new_collation.toml:26]:
```suggestion
```

This option have been removed.

File: cmd/explaintest/run-tests.sh [hunk 2]
```diff
@@ -22,6 +22,7 @@ importer=""
 tidb_server=""
 portgenerator=""
 explain_test_log="./explain-test.out"
+explain_test_disable_new_collation_log="./explain-test-disable-new-collation.out"
```
Comment (tangenta) [inline cmd/explaintest/run-tests.sh:25]: Why not use `>>` redirecting the logs into `./explain-test.out`?

So you don't have to update [the CI script](https://github.com/PingCAP-QE/ci/blob/main/jenkins/pipelines/ci/tidb/tidb_ghpr_check.groovy#L175).

**Review by tangenta — APPROVED** [review]

**Review by xiongjiwei — APPROVED** [review]

**Comment by wjhuang2016** [comment]
@ti-chi-bot

**Comment by wjhuang2016** [comment]
@ti-chi-bot

**Pull request merged**